nova_plugin/server.py

```diff
@@ -548,11 +548,21 @@ def attach_volume(nova_client, cinder_client, **kwargs):
         server_id,
         volume_id,
         device if device != 'auto' else None)
-    volume.wait_until_status(cinder_client=cinder_client,
-                             volume_id=volume_id,
-                             status=volume.VOLUME_STATUS_IN_USE)
-    if device == 'auto':
+    try:
+        vol, wait_succeeded = volume.wait_until_status(
+            cinder_client=cinder_client,
+            volume_id=volume_id,
+            status=volume.VOLUME_STATUS_IN_USE
+        )
+        if not wait_succeeded:
+            raise RecoverableError(
+                'Waiting for volume status {0} failed - detaching volume and '
+                'retrying..'.format(volume.VOLUME_STATUS_IN_USE))
+    except RecoverableError:
```

Comment:
2) not sure what you mean, since I don't see any new commits; perhaps i didn't explain myself well enough - its true that if "wait_succeeded" is false, the exception will get caught - however, the "wait_until_status" call may raise an error which isn't a RecoverableError in itself but is "recoverable" in the Cloudify sense (i.e. it'll get retried), which is why line 561 should probably be "except Exception" rather than "except RecoverableError" - and then NonRecoverableErrors should be filtered out.

regarding (1), you're correct, however I think that's just one more problem we need to address rather than let it exist in another place as well; I think that if a cleanup/revert after a failed operation fails in itself, we should raise a NonRecoverableError, as the environment is in an inconsistent state. What do you think?